yt_dlp/extractor/newgrounds.py

```diff
@@ -73,38 +102,16 @@ def _real_extract(self, url):
                 'format_id': 'source',
                 'quality': 1,
             }]
-
-            max_resolution = int_or_none(self._search_regex(
-                r'max_resolution["\']\s*:\s*(\d+)', webpage, 'max resolution',
-                default=None))
-            if max_resolution:
-                url_base = media_url.rpartition('.')[0]
-                for resolution in (360, 720, 1080):
-                    if resolution > max_resolution:
-                        break
-                    formats.append({
-                        'url': '%s.%dp.mp4' % (url_base, resolution),
-                        'format_id': '%dp' % resolution,
-                        'height': resolution,
-                    })
         else:
-            video_id = int_or_none(self._search_regex(
-                r'data-movie-id=\\"([0-9]+)\\"', webpage, ''))
-            if not video_id:
-                raise ExtractorError('Could not extract media data')
-
-            url_video_data = 'https://www.newgrounds.com/portal/video/%s' % video_id
-            headers = {
+            json_video = self._download_json('https://www.newgrounds.com/portal/video/' + media_id, media_id, headers={
                 'Accept': 'application/json',
                 'Referer': url,
                 'X-Requested-With': 'XMLHttpRequest'
-            }
-            json_video = self._download_json(url_video_data, video_id, headers=headers, fatal=False)
+            }, fatal=False)
             if not json_video:
```

Comment:
This is not a suggestion or comment but a question

Why not use `fatal=True`?
```py
            if fatal:
                raise ExtractorError(errmsg, sys.exc_info()[2], cause=err)
```
this how the `fatal` argument is used.

Reply:
You r right. This code was written in 2014 and last updated seriously in 2017. I assume the fatal keyword didnt exist back then. But we should be using it now

Reply:
I don't know it was like that i forgot to change it.

Reply:
Here's the func
```py
def _download_json(
            self, url_or_request, video_id, note='Downloading JSON metadata',
            errnote='Unable to download JSON metadata', transform_source=None,
            fatal=True, encoding=None, data=None, headers={}, query={},
            expected_status=None)
```
So you can add the error note in `errnote`.

Reply:
I mean there is nothing wrong with default error note there is no need for custom ```errnote```.